### PR TITLE
Move `Command` infrastructure to parser 

### PR DESCRIPTION
**Pull request by 4tXJ7f**
This commit moves the `Command` infrastructure to the parser (to
`src/parser/api/cpp`) and moves it to the `cvc5::parser` namespace. The
`Command`-related files are compiled with `-D__BUILDING_CVC5LIB`,
because they need access to the internal printers. The commit refactors
the internal code to not depend on `Command` anymore. To do this, the
commit refactors printing derived classes of `CommandStatus` to follow a
similar structure to printing commands (so that we can also print
command statuses internally without creating instances of
`Commandstatus`). This is a step towards a parser API.

### CODE REVIEW

File: src/api/cpp/cvc5.h [hunk 1]
```diff
@@ -350,7 +352,7 @@ class Datatype;
  */
 class CVC5_EXPORT Sort
 {
-  friend class cvc5::Command;
+  friend class parser::Command;
```
Comment (mpreiner) [inline src/api/cpp/cvc5.h:355]: Do we still need the friend declarations for Command?

File: src/printer/tptp/tptp_printer.cpp [hunk 2]
```diff
@@ -61,6 +54,34 @@ void TptpPrinter::toStream(std::ostream& out, const smt::Model& m) const
       << endl;
 }
 
+void TptpPrinter::toStreamCmdSuccess(std::ostream& out) const
+{
+  getPrinter(Language::LANG_SMTLIB_V2_6)->toStreamCmdSuccess(out);
```
Comment (mpreiner) [inline src/printer/tptp/tptp_printer.cpp:59]: We use the `Language::LANG_SMTLIB_V2_6` printer for TPTP?

File: src/api/cpp/cvc5.h (hunk 1 above)
Comment (4tXJ7f) [reply src/api/cpp/cvc5.h:355]: Yes, I think that we currently still need it, but I am working on making that obsolete.

File: src/printer/tptp/tptp_printer.cpp (hunk 2 above)
Comment (4tXJ7f) [reply src/printer/tptp/tptp_printer.cpp:59]: Yeah, our TPTP printer doesn't really do much...

**Review by mpreiner — APPROVED** [review]

**Pull request merged**